azurerm/internal/services/storage/tests/resource_arm_storage_sync_cloud_endpoint_test.go

```diff
@@ -0,0 +1,165 @@
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMCloudEndpoint_requiresImport),
+		},
+	})
+}
+
+func testCheckAzureRMStorageSyncCloudEndpointExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("storage Sync Cloud Endpoint not found: %s", resourceName)
+		}
+
+		id, err := parsers.SyncCloudEndpointID(rs.Primary.Attributes["id"])
+		if err != nil {
+			return err
+		}
+
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Storage.CloudEndpointsClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.StorageSyncName, id.StorageSyncGroup, id.Name); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Storage Sync Cloud Endpoint %q (Storage Sync Group %q / Storage Sync Service Name %q / Resource Group %q) does not exist", id.Name, id.ResourceGroup, id.StorageSyncName, id.ResourceGroup)
+			}
+			return fmt.Errorf("bad: Get on CloudEndpointsClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMStorageSyncCloudEndpointDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).Storage.CloudEndpointsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_storage_sync_cloud_endpoint" {
+			continue
+		}
+
+		id, err := parsers.SyncCloudEndpointID(rs.Primary.Attributes["id"])
+		if err != nil {
+			return err
+		}
+
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.StorageSyncName, id.StorageSyncGroup, id.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Get on CloudEndpointsClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+	return nil
+}
+
+func testAccAzureRMCloudEndpoint_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_storage_sync" "test" {
+  name                = "acctest-storagesync-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+}
+
+resource "azurerm_storage_sync_group" "test" {
+  name            = "acctest-StorageSyncGroup-%[1]d"
+  storage_sync_id = azurerm_storage_sync.test.id
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "accstr%[3]s"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_storage_share" "test" {
+  name                 = "acctest-share-%[1]d"
+  storage_account_name = azurerm_storage_account.test.name
+
+  lifecycle {
```

Comment:
Can we fix the resource so this isn't required?

Reply:
Sure.